### PR TITLE
Allow Flock and Silicons to hear each other's radio speech

### PR DESCRIPTION
**Pull request by FlameArrow57**
[GAME MODES][FEATURE][REWORK]

## About the PR 
This PR just enables an unimplemented/unfinished mechanic for Flock where Flock can hear robot talk (:s) and silicons can hear Flock speak.

Also reworks it so that Flock can always hear robot talk with slight garble and silicons can only hear Flock with low clarity and only when they've reached a certain amount of compute, with a chance of not hearing it.

## Why's this needed? 
Planned Flock feature

### CODE REVIEW

File: code/procs/mobprocs/chatprocs.dm [hunk 1]
```diff
@@ -1090,7 +1091,7 @@
 
 		if((isflockmob(M)) || (M.client.holder && !M.client.player_mode) || (isobserver(M) && !(istype(M, /mob/dead/target_observer/hivemind_observer))))
 			thisR = rendered
-		if(flock?.snooping && M.client && M.robot_talk_understand)
+		if(M.client && (M.robot_talk_understand || istype(M, /mob/living/intangible/aieye)))
```
Comment (TobleroneSwordfish) [inline code/procs/mobprocs/chatprocs.dm:1094]:
```suggestion
		if(M.robot_talk_understand || istype(M, /mob/living/intangible/aieye))
```
You're iterating through clients, so no need to check if it exists. Also is the `aieye` check necessary? The `robot_talk` proc doesn't have it.

Comment (FlameArrow57) [reply code/procs/mobprocs/chatprocs.dm:1094]: Yeah it is, not sure why

Comment (pali6) [reply code/procs/mobprocs/chatprocs.dm]: I think the mainframe relays the robot talk to the AI eye so it doesn't have the var set? Could be worth looking into if setting the var on aieye doubles the messages or not.

**Review by pali6 — APPROVED** [review]
Looks good to me at a glance and you flock people know what you are doing I think.

File: code/datums/flock/flockunlockable.dm [hunk 2]
```diff
@@ -41,7 +41,7 @@ ABSTRACT_TYPE(/datum/unlockable_flock_structure)
 	structType = /obj/flock_structure/relay
 
 	check_unlocked()
-		return ..() || (src.my_flock.total_compute() > 500 && !src.my_flock.relay_in_progress_or_finished)
+		return ..() || (src.my_flock.total_compute() > FLOCK_RELAY_COMPUTE_COST && !src.my_flock.relay_in_progress_or_finished)
```
Comment (FlameArrow57) [inline code/datums/flock/flockunlockable.dm:44]:
```suggestion
		return ..() || (src.my_flock.total_compute() >= FLOCK_RELAY_COMPUTE_COST && !src.my_flock.relay_in_progress_or_finished)
```
Merge resolve

**Comment by FlameArrow57** [comment]
!merge_upstream

**Pull request merged**